boards/xtensa/esp32/doc/esp32.rst

```diff
@@ -165,6 +163,30 @@ pinout.  Power to the ESP32 board should be provided by a USB cable.
 | IO15       | TDO       |
 +------------+-----------+
 
+Once the device is connected, you should be able to connect with (for
+a DevKitC board, replace with esp32-wrover.cfg for WROVER):
+
+.. code-block:: console
+
+    cd ~/esp/openocd-esp32
+    sudo src/openocd -f interface/ftdi/flyswatter2.cfg -c 'set ESP32_ONLYCPU 1' -c 'set ESP32_RTOS none' -f board/esp-wroom-32.cfg -s tcl
+
+The ESP32_ONLYCPU setting is critical: without it OpenOCD will present
+only the "APP_CPU" via the gdbserver, and not the "PRO_CPU" on which
+Zephyr is running.  It's currently unexplored as to whether the CPU
+can be switched at runtime or if breakpoints can be set for
+either/both.
+    
+Now you can connect to openocd with gdb and point it to the OpenOCD
+gdbserver running (by default) on localhost port 3333.  Note that you
+must use the gdb distributed with the ESP-32 SDK.  Builds off of the
+FSF mainline get inexplicable protocol errors when connecting.
+    
+.. code-block:: console
+
+    ~/esp/xtensa-esp32-elf/bin/xtensa-esp32-elf-gdb outdir/esp32/zephyr.elf
```

Comment:
I use the following command line: `xtensa-esp32-elf-gdb -ex 'target remote 127.0.0.1:3333' outdir/esp32/zephyr.elf`. It's the same thing, but easier to find in the shell history.

Reply:
Me too.  But it's harder to explain in docs. :)  This matches the FSF docs on gdb as to the default mechanism I'm pretty sure.